### PR TITLE
Update build instruction for Go 1.12+

### LINKED ISSUE #227: Outdated "build from source" instructions
The [instruction to get the source code](https://podman.io/getting-started/installation#get-source-code) sounds strange for a project that uses `go` modules. Is it still actual? What is the minimal `go` version for building `podman` 1.9.x?

### PR DESCRIPTION
**Pull request by abitrolly**
Fixes #227

### CODE REVIEW

File: getting-started/installation.md [hunk 1]
```diff
@@ -526,23 +526,11 @@ apt-get install -y \
 
 ### Get Source Code
 
-As with other Go projects, Podman must be cloned into a directory structure like:
```
Comment (rhatdan) [inline getting-started/installation.md:529]: Why did you remove the GOPATH section?

Comment (abitrolly) [reply getting-started/installation.md:529]: Because `podman` uses go modules and doesn't support `go` less than 1.11

**Comment by rhatdan** [comment]
@vrothberg @nalind @TomSweeneyRedHat @mheon PTAL

**Comment by TomSweeneyRedHat** [comment]
@rhatdan, fwiw, running the process on a new fresh VM to check it out.  I ran into a few issues, but they were all self-inflicted so far.  Will add a final comment later today.

**Comment by TomSweeneyRedHat** [comment]
@abitrolly sorry for the delay, finally had some time to run through this on a new VM and then created some self-inflicted problems while doing so.  Anyway, it all looks good and is simpler than the previous process.  Thanks for running this down and creating the PR!
LGTM
